dotcom-rendering/src/components/ArticleMeta.apps.tsx

```diff
@@ -0,0 +1,338 @@
+	grid-template-rows: auto auto auto auto auto;
+	grid-template-columns: 10px auto 1fr auto 10px;
+	grid-template-areas:
+		' . . . . .'
+		'. avatar byline comment-count .'
+		'. avatar byline comment-count .'
+		' . . . . .'
+		'. dateline dateline dateline .';
+
+	${from.mobileLandscape} {
+		grid-template-columns: 20px auto 1fr auto 20px;
+	}
+
+	${from.phablet} {
+		grid-template-columns: 0px auto 1fr auto 0px;
+	}
+`;
+
+const metaContainerMargins = css`
+	${until.phablet} {
+		margin-left: -20px;
+		margin-right: -20px;
+	}
+	${until.mobileLandscape} {
+		margin-left: -10px;
+		margin-right: -10px;
+	}
+`;
+
+const MetaGridAvatar = ({ children }: { children: React.ReactNode }) => (
+	<div
+		css={css`
+			grid-area: avatar;
+			width: 60px;
+			height: 60px;
+			margin-top: ${space[2]}px;
+			margin-right: ${space[3]}px;
+			margin-bottom: ${space[3]}px;
+			margin-left: 0px;
+			${from.leftCol} {
+				width: 140px;
+				height: 140px;
+			}
+		`}
+	>
+		{children}
+	</div>
+);
+
+const MetaGridByline = ({
+	children,
+	isComment,
+}: {
+	children: React.ReactNode;
+	isComment: boolean;
+}) => (
+	<div
+		css={css`
+			grid-area: byline;
+			width: auto;
+			margin-top: ${space[2]}px;
+			margin-bottom: ${space[3]}px;
+
+			address > div {
+				padding-bottom: 0px;
+			}
+
+			address > div > svg:first-of-type {
+				display: block;
+				margin-top: ${space[2]}px;
+				${!isComment && `margin-bottom: ${space[2]}px;`}
+			}
+		`}
+	>
+		{children}
+	</div>
+);
+
+const MetaGridCommentCount = ({
+	children,
+	isPicture,
+	isComment,
+	isImmersive,
+	isAnalysis,
+	isLiveBlog,
+}: {
+	children: React.ReactNode;
+	isPicture: boolean;
+	isComment: boolean;
+	isImmersive: boolean;
+	isAnalysis: boolean;
+	isLiveBlog: boolean;
+}) => (
+	<div
+		data-print-Content="hide"
+		css={css`
+			grid-area: comment-count;
+			border-left: 1px solid
+				${isLiveBlog
+					? 'rgba(255, 255, 255, 0.4)'
+					: themePalette('--article-border')};
+			padding-top: ${space[2]}px;
+			padding-left: ${space[2]}px;
+			${(isComment || isImmersive || isAnalysis || isLiveBlog) &&
+			`padding-bottom: ${space[3]}px;`}
+			${isPicture && 'margin-top: -4px;'}
+
+			${from.mobileLandscape} {
+				padding-top: ${space[2]}px;
+				padding-left: ${space[4]}px;
+			}
+
+			${from.phablet} {
+				padding-right: ${space[4]}px;
+			}
+		`}
+	>
+		{children}
+	</div>
+);
+
+const MetaGridDateline = ({
+	children,
+	isImmersiveOrAnalysisWithMultipleAuthors,
+}: {
+	children: React.ReactNode;
+	isImmersiveOrAnalysisWithMultipleAuthors: boolean;
+}) => (
+	<div
+		css={css`
+			grid-area: ${isImmersiveOrAnalysisWithMultipleAuthors
+				? 'byline'
+				: 'dateline'};
+			margin-top: ${space[2]}px;
+			margin-bottom: ${space[4]}px;
+		`}
+	>
+		{children}
+	</div>
+);
+
+const stretchLines = ({
+	isPicture,
+	isLiveBlog,
+}: {
+	isPicture: boolean;
+	isLiveBlog: boolean;
+}) => css`
+	grid-column: 1 / -1;
+
+	${(isPicture || isLiveBlog) &&
+	`grid-column: 2 / -2;
+	`}
+`;
```

Comment:
It’s best to extract this to two static styles that are composed in the css prop with an array:

```tsx
const strechLines = css`
    grid-column: 1 / -1;
`

const strechLinesLess = css`
    grid-column: 2 / -2;
`

const Component = () => (
  <div css={isPicture || isLiveBlog ? strechLineLess : strechLine}>Content</div>
)
```